app/process/recovery/recovery.go

```diff
@@ -235,61 +234,61 @@ func (r Recovery) recoverEthereumTXMessage(tm encoding.TopicMessage) error {
 	return nil
 }
 
+type TransferMessageKey struct {
+	TransactionId string
+	EthAddress    string
+	Amount        string
+	Fee           string
+	GasPriceWei   string
+	Erc20Address  string
+}
+
 func (r Recovery) processUnfinishedOperations() error {
-	// TODO messagesRepo.getUnprocessedMessages() <- should return all Messages whose TX ID is status INITIAL or RECOVERED
-	//unprocessed, err := r.transactionsRepo.GetSkippedOrInitialTransactionsAndMessages()
-	//if err != nil {
-	//	r.logger.Fatalf("Failed to get all unprocessed messages. Error: %s", err)
-	//	return err
-	//}
-
-	// Combine all Messages records into map({txId, amount, fee, gasPrice, etc}, []signatures)
-	// Iterate all keys
-	// 	If validator has not signed -> sign and submit
-	//for txn, txnSignatures := range unprocessed {
-	//hasSubmittedSignature, ctm := r.hasSubmittedSignature(txn, txnSignatures)
-	//
-	//if !hasSubmittedSignature {
-	//	r.logger.Infof("Validator has not yet submitted signature for Transaction with ID [%s]. Proceeding now...", txn)
-	//	// TODO
-	//	err = r.transfersService.VerifyFee(ctm)
-	//	if err != nil {
-	//		r.logger.Errorf("Fee validation failed for TX [%s]. Skipping further execution", transferMsg.TransactionId)
-	//	}
-	//
-	//	signature, err := r.transfersService.ValidateAndSignTxn(ctm)
-	//	if err != nil {
-	//		r.logger.Errorf("Failed to Validate and Sign TransactionID [%s]. Error [%s].", txn, err)
-	//	}
-	//
-	//	_, err = r.transfersService.HandleTopicSubmission(ctm, signature)
-	//	if err != nil {
-	//		return errors.New(fmt.Sprintf("Could not submit Signature [%s] to Topic [%s] - Error: [%s]", signature, r.topicID, err))
-	//	}
-	//	r.logger.Infof("Successfully Validated")
-	//}
-	//}
+	unprocessedMessages, err := r.messagesRepo.GetUnprocessedMessages()
+	if err != nil {
+		r.logger.Fatalf("Failed to get all unprocessedMessages messages. Error: %s", err)
+		return err
+	}
+
+	signatureMessagesMap := make(map[*TransferMessageKey][]string)
+
+	for _, txnMessage := range unprocessedMessages {
+		key := &TransferMessageKey{
+			TransactionId: txnMessage.TransactionId,
+			EthAddress:    txnMessage.EthAddress,
+			Amount:        txnMessage.Amount,
+			Fee:           txnMessage.Fee,
+			GasPriceWei:   txnMessage.GasPriceWei,
+			Erc20Address:  txnMessage.ERC20ContractAddress,
+		}
+		signatureMessagesMap[key] = append(signatureMessagesMap[key], txnMessage.Signature)
+	}
+
+	for txn, txnSignatures := range signatureMessagesMap {
```

Comment:
nit: I would say that a better name for the variable is `transfer` or `tr` and not `txn